### PR TITLE
Use unescaped values for mustache strings

### PR DESCRIPTION
**Pull request by teallarson**
## Change description

Initially, we thought there was something wrong with SQLite's escaping.  It turns out, it was mustachejs returning escaped strings in Calculated Property functions.

Mustache variables automatically escape HTML. However, with calculated properties, we actually want to be using the unescaped HTML value of a string.  That way, if a property value has an HTML-escapable character like an apostrophe, it gets rendered correctly.

Because of how our system parses in the values for mustache strings, we'll start by defaulting to triple brackets (ie: `{{{ firstName }}}` everywhere.

I believe we will want to revisit this and simply keep whatever bracket notation a user inputs by making our parser more sophisticated.  For now, we'll start here.

O'Donoghue in `fullName` property used to be one of the culprits, now renders fine:

Screen Shot 2021-12-10 at 4 55 39 PM

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:
- Users who have properties that have an option with mustache variables will be replaced by the triple bracket version (both in the database when Grouparoo is running and within the actual config files if Config mode is run)
- All property options (calculated property or query source property) with mustache variables will now be saved as the triple bracket notation version of whatever is entered (double or triple)

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/src/modules/ops/property.ts [hunk 1]
```diff
@@ -134,7 +134,7 @@ export namespace PropertyOps {
     for (const key in ruleOptions) {
       const mustacheString = ruleOptions[key];
       const mustacheVariables: string[] = Mustache.parse(mustacheString)
-        .filter((chunk) => chunk[0] === "name")
+        .filter((chunk) => chunk[0] === "&")
```
Comment (teallarson) [inline core/src/modules/ops/property.ts:137]: This was the missing piece that had broken the DAG this morning!  When one uses the triple bracket, `Mustache.parse`'s response looks different.  The ampersand tells mustache to render the value unescaped.

Comment (evantahler) [reply core/src/modules/ops/property.ts:137]: That's 🍌s

Some notes here - https://stackoverflow.com/questions/22910428/mustache-globally-disable-html-escaping/22910523

File: plugins/@grouparoo/calculated-property/__tests__/import/import-property.ts [hunk 2]
```diff
@@ -55,28 +55,35 @@ describe("calculated-property/recordProperty", () => {
...
     }`;
     const value = await getPropertyValue(fn);
     expect(value[0]).toEqual(`hi Mario`);
   });
+  test("it evaluates string properties with escapable characters as expected", async () => {
+    const fn = `() => {
+        return "hi {{{firstName}}} {{{ lastName }}}"
+    }`;
+    const value = await getPropertyValue(fn);
+    expect(value[0]).toEqual(`hi Mario O'Donnell`);
```
Comment (evantahler) [inline plugins/@grouparoo/calculated-property/__tests__/import/import-property.ts:68]: 👍 good testing

File: plugins/@grouparoo/snowflake/__tests__/query-import/import-property.ts [hunk 3]
```diff
@@ -50,43 +50,43 @@ describe("snowflake/query/recordProperty", () => {
...
   test("can run a integer query to get a string", async () => {
-    const sql = "SELECT first_name FROM profiles WHERE id = {{ userId }}";
+    const sql = "SELECT first_name FROM profiles WHERE id = {{{ userId }}}";
     const value = await getPropertyValue(sql);
     expect(value).toEqual(["Erie"]);
   });
 
   test("can run a integer query to get a float", async () => {
-    const sql = "SELECT ltv FROM profiles WHERE id = {{ userId }}";
+    const sql = "SELECT ltv FROM profiles WHERE id = {{{ userId }}}";
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/__tests__/query-import/import-property.ts:59]: I'm curious - would these queries have still worked with the `{{}}`?

**Review by evantahler — APPROVED** [review]

File: core/src/modules/ops/property.ts (hunk 1 above)
Comment (teallarson) [reply core/src/modules/ops/property.ts:137]: Interesting!  I had to dig around in their github repo to find this rule 😅

File: plugins/@grouparoo/snowflake/__tests__/query-import/import-property.ts (hunk 3 above)
Comment (teallarson) [reply plugins/@grouparoo/snowflake/__tests__/query-import/import-property.ts:59]: It does!  Just checked.  Core converts it all correctly and figures it out.

**Pull request merged**